apps/nxls/src/main.ts

```diff
@@ -233,6 +239,23 @@ connection.onRequest(NxWorkspaceRequest, async ({ reset }) => {
   return nxWorkspace(WORKING_PATH, lspLogger, reset);
 });
 
+connection.onRequest(
+  NxGeneratorsRequest,
+  async (args: {
+    projects?: WorkspaceProjects;
+    options?: GetGeneratorsOptions;
+  }) => {
+    if (!WORKING_PATH) {
+      return new ResponseError(
+        1000,
+        'Unable to get Nx info: no workspace path'
+      );
+    }
+
+    return getGenerators(WORKING_PATH, args.projects, args.options);
```

Comment:
You don't need to pass projects as an arg to the lsp. It can get it by itself with `nxWorkspace`

```suggestion
const {
      workspace: { projects },
    } = await nxWorkspace(WORKING_PATH, lspLogger);

    return getGenerators(WORKING_PATH, projects, args.options);
```

Reply:
done, i just get the projects inside the readCollections function now. So I don't have to pass it around needlessly